src/Compiler/Checking/CheckBasics.fs

```diff
@@ -344,7 +346,8 @@ type TcFileState =
           tcSimplePats,
           tcSequenceExpressionEntry,
           tcArrayOrListSequenceExpression,
-          tcComputationExpression) =
+          tcComputationExpression,
+          ?sourceText: ISourceText option) =
```

Comment:
I don't think this needs to be doubly-optional.

With just

```suggestion
          ?sourceText: ISourceText) =
```

a caller can still pass in an option explicitly like `Create (…, ?sourceText=sourceTextOpt)` if they need.